core/src/test/java/com/spotify/missinglink/FeatureTest.java

```diff
@@ -209,8 +233,61 @@ public void testNoConflictWithCovariantReturnType() throws Exception {
     assertThat(conflictChecker
         .check(artifact,
             ImmutableList.of(artifact),
+            ImmutableList.of(artifact)
+        )).isEmpty();
+  }
+
+  @org.junit.Test
+  public void testNoConflictWithStaticCall() throws Exception {
+    final DeclaredMethod methodOnlyInSuper = newMethod(true, INT, "foo").build();
+    final DeclaredClass superClass =
+            newClass("com/super").methods(methodMap(methodOnlyInSuper)).build();
+
+    final CalledMethod methodCall = newCall(superClass, methodOnlyInSuper, true, false);
+    final DeclaredMethod mainMethod = newMethod(true, VOID, "main", array(STRING))
+            .methodCalls(ImmutableSet.of(methodCall))
+            .fieldAccesses(ImmutableSet.of())
+            .build();
+
+    final DeclaredClass mainClass = newClass("com/Main").methods(methodMap(mainMethod)).build();
+
+    final Artifact artifact = newArtifact("art", superClass, mainClass);
+
+    assertThat(conflictChecker.check(artifact,
             ImmutableList.of(artifact),
-            ImmutableList.of())).isEmpty();
+            ImmutableList.of(artifact)
+    )).isEmpty();
+  }
+
+  @org.junit.Test
+  public void testConflictWithStaticToVirtualCall() throws Exception {
+    final DeclaredMethod methodOnlyInSuper = newMethod(false, INT, "foo").build();
+    final DeclaredClass superClass =
+            newClass("com/super").methods(methodMap(methodOnlyInSuper)).build();
+
+    final CalledMethod methodCall = newCall(superClass, methodOnlyInSuper, true, false);
+    final DeclaredMethod mainMethod = newMethod(true, VOID, "main", array(STRING))
+            .methodCalls(ImmutableSet.of(methodCall))
+            .fieldAccesses(ImmutableSet.of())
+            .build();
+
+    final DeclaredClass mainClass = newClass("com/Main").methods(methodMap(mainMethod)).build();
+
+    final Artifact artifact = newArtifact("art", superClass, mainClass);
+
+    final Conflict expectedConflict = new ConflictBuilder()
+            .dependency(dependency(mainClass.className(), mainMethod, methodCall))
+            .reason("Method not found: com.super.foo()")
+            .category(ConflictCategory.METHOD_SIGNATURE_NOT_FOUND)
+            .usedBy(artifact.name())
+            .existsIn(artifact.name())
+            .build();
+
+    assertEquals(Arrays.asList(expectedConflict),
+            conflictChecker.check(artifact,
+                    ImmutableList.of(artifact),
+                    ImmutableList.of(artifact)
+            ));
```

Comment:
Should there also be a test that verifies that an attempt at doing a 'static' call to a non-static method raises a warning? I haven't tested it, but suppose it would fail?